File: searching_algo/linear_search.py
# Linear Search

def linear_search(num_list, key):
    for i in range(len(num_list)):
        if num_list[i] == key:
            return i + 1 
    return False

numbers = [1, 2, 3, 4, 5, 6, 7, 8, 9, 10]
print(linear_search(numbers, 6))
print(linear_search(numbers, 20))

File: searching_algo/linear_search.c
// Linear Search
// 순차 탐색

#include <stdio.h>
#define MAX_SIZE 100

int list[MAX_SIZE];		// List 선언

// fuction for making list
int make_list(int first, int last) {

	int index = 0;
	
	for (int i = first; i <= last; i++) {
		list[index] = i;
		index++;
	}
	
	return 0;
}

// function for linear search
int linear_search(int key) {

	int index;

	for (index = 0; index <= MAX_SIZE; index++) {
		if (list[index] == key) {

			printf("Index of %d : No.%d\n", key, index);

			break;
		}
	}
	
	return 0;
}

int main() {

	make_list(1, 10);
	linear_search(4);

	make_list(50, 128);
	linear_search(88);

}
